integration-tests/validation.py

```diff
@@ -74,8 +74,12 @@ def validate_status_messages(model):
     action = await unit.run_action('microbot', replicas=3)
     await action.wait()
     assert action.status == 'completed'
-    # TODO: wait for pods running
-    # TODO: test that we can reach the ingress endpoint
+    for i in range(60):
+        resp = requests.get('http://' + action.data['results']['address'])
```

Comment:
This blocks the event loop. It would be better to use aiohttp, or run this with `run_in_executor`.